src/services/search.ts

```diff
@@ -0,0 +1,29 @@
+// parent search. holds logic for splitting between different services
+
+import * as play from 'play-dl';
+import { PlayValidation } from '../types/definitions';
+
+// TODO: handle search for other sources. for now, default to yt_videos only
+export const search = async (
+    query: string,
+): Promise<play.YouTubeVideo | null> => {
```

Comment:
unless the API expects a null, prefer undefined.

Reply:
Curious, why undefined? I don't like undefined, since it could be naturally set. I think that `null` implies that a developer intentionally wants to return nothing for this value.

Reply:
https://stackoverflow.com/questions/6604749/what-reason-is-there-to-use-null-instead-of-undefined-in-javascript